src/main/java/ch/uzh/ifi/hase/soprafs22/service/GameService.java

```diff
@@ -239,14 +246,78 @@ public Game getGameFromRandomUser(User user) {
     // limit page size to 100
     int pageSize = (numOfGames < 100) ? numOfGames.intValue() : 100;
     int pageIndex = rand.nextInt(pageSize);
-    // only get one game
-    PageRequest pageRequest = PageRequest.of(pageIndex, 1);
+    // all games it found
+    PageRequest pageRequest = PageRequest.of(pageIndex, pageSize);
+
+    // get the page with the games
+    Page<Game> somePage = gameRepository.getOtherUserWithActiveGameThatWasNotPlayedOn(
+      pageRequest, 
+      user.getId(), 
+      user,
+      user.getGender(), 
+      minAgeDate, 
+      maxAgeDate, 
+      user.getBlockedUsers(),
+      user.getMatchedUsers()
+    );
+
+    // subset the page of users to only users that have a haversine distance of less than user.getMaxRange()
+    List<Game> games = somePage.getContent();
+    Game matchingGame = null;
+
+    // go over all games that match the other criteria
+    for(Game game : games){
+      // for each game get the user that the game belongs to
+      User gameUser = game.getUser();
+
+      // calculate the distance between the two users
+      double distance = haversineDistance(
+        user.getLatitude(), 
+        user.getLongitude(), 
+        gameUser.getLatitude(), 
+        gameUser.getLongitude()
+      );
+
+      // if the distance is less than the max range, add the game to the list
+      if(distance <= user.getMaxRange()){
+        matchingGame = game;
+        break;
```

Comment:
@TwoDigitsOneNumber Why did you break the loop and not just instantly return the game?